plugins/buildClusters/create.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+
+const boom = require('boom');
+const urlLib = require('url');
+const validationSchema = require('screwdriver-data-schema');
+
+module.exports = () => ({
+    method: 'POST',
+    path: '/buildclusters',
+    config: {
+        description: 'Create a build cluster',
+        notes: 'Create a specific build cluster',
+        tags: ['api', 'buildclusters'],
+        auth: {
+            strategies: ['token'],
+            scope: ['user', '!guest']
+        },
+        plugins: {
+            'hapi-swagger': {
+                security: [{ token: [] }]
+            }
+        },
+        handler: (request, reply) => {
+            const buildClusterFactory = request.server.app.buildClusterFactory;
+            const userFactory = request.server.app.userFactory;
+            const scm = buildClusterFactory.scm;
+            const username = request.auth.credentials.username;
+            const scmContext = request.auth.credentials.scmContext;
+            const scmOrganizations = request.payload.scmOrganizations;
+            const payload = {
+                name: request.payload.name,
+                scmOrganizations,
+                managedByScrewdriver: request.payload.managedByScrewdriver,
+                maintainer: request.payload.maintainer,
+                description: request.payload.description,
+                isActive: request.payload.isActive,
+                weightage: request.payload.weightage,
+                scmContext
+            };
+
+            // Check permissions
+            // Must be Screwdriver admin to add Screwdriver build cluster
+            if (payload.managedByScrewdriver) {
```

Comment:
Should we should make this also an `admin` action so that SD Cluster admins can handle end-to-end onboarding and not expose a custom cluster without everything setup/

Reply:
thought about that, but that means the admin need to manually check if the build cluster maintainer is actually admins for all the github orgs he specified in scmOrganizations.